File: RoomReservationBackend/Models/Reservation.cs
using System;

namespace RoomReservationBackend.Models
{
    public class Reservation
    {
        public int ReservationId { get; set; }
        public int RoomId { get; set; }
        public int UserId { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public bool IsApproved { get; set; }

        // Foreign keys
        public Room? Room { get; set; }
        public User? User { get; set; }
    }
}

File: room_reservation_app/lib/Models/reservation.dart
class Reservation {
  final String startTime;
  final String endTime;
  final int roomId;

  Reservation({
    required this.startTime,
    required this.endTime,
    required this.roomId,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation(
      startTime: json['startTime'],
      endTime: json['endTime'],
      roomId: json['roomId'],
    );
  }
}
